src/indra_cogex/client/queries.py

```diff
@@ -1740,6 +1794,1447 @@ def _filter_out_medscan_evidence(
     ]
 
 
+# Cell marker functions
+@autoclient()
+def get_markers_for_cell_type(
+    cell_type: Tuple[str, str], *, client: Neo4jClient
+) -> Iterable[Node]:
+    """Return the markers associated with the given cell type.
+
+    Parameters
+    ----------
+    client :
+        The Neo4j client
+    cell_type :
+        The cell type to query (e.g., ("cl", "0000020"))
+
+    Returns
+    -------
+    :
+        The markers (genes) associated with the cell type
+    """
+    cell_id = f"{cell_type[0]}:{cell_type[1]}".lower()
+    query = """
+    MATCH (c:BioEntity {id: $cell_id})-[r:has_marker]->(m:BioEntity)
+    RETURN m
+    """
+    return client.query_nodes(query, cell_id=cell_id)
+
+
+@autoclient()
+def get_cell_types_for_marker(
+    marker: Tuple[str, str], *, client: Neo4jClient
+) -> Iterable[Node]:
+    """Return the cell types associated with the given marker.
+
+    Parameters
+    ----------
+    client :
+        The Neo4j client
+    marker :
+        The marker (gene) to query (e.g., ("HGNC", "11337"))
+
+    Returns
+    -------
+    :
+        The cell types associated with the marker
+    """
+    marker_id = f"{marker[0]}:{marker[1]}".lower()
+    query = """
+    MATCH (c:BioEntity)-[r:has_marker]->(m:BioEntity {id: $marker_id})
+    RETURN c
+    """
+    return client.query_nodes(query, marker_id=marker_id)
+
+
+@autoclient()
+def is_marker_for_cell_type(
+    marker: Tuple[str, str], cell_type: Tuple[str, str], *, client: Neo4jClient
+) -> List[Any]:
+    """Return True if the marker is associated with the given cell type.
+
+    Parameters
+    ----------
+    client :
+        The Neo4j client
+    marker :
+        The marker to query
+    cell_type :
+        The cell type to query
+
+    Returns
+    -------
+    :
+        True if the marker is associated with the cell type
+    """
+    marker_id = f"{marker[0]}:{marker[1]}".lower()
+    cell_id = f"{cell_type[0]}:{cell_type[1]}".lower()
+    query = """
+    MATCH (c:BioEntity {id: $cell_id})-[r:has_marker]->(m:BioEntity {id: $marker_id})
+    RETURN COUNT(r) > 0 as exists
+    """
+    return client.query_tx(query, cell_id=cell_id, marker_id=marker_id, squeeze=True)[0]
+
+
+# HPOA Functions
+@autoclient()
+def get_phenotypes_for_disease(
+    disease: Tuple[str, str], *, client: Neo4jClient
```

Comment:
See if you can find an example for `disease` that returns a result for this function in the API docs, right now the result is empty.

Reply:
FIxed in 4640b451